tasks/main.yml

```diff
@@ -223,6 +223,12 @@
     inventory_mode: "{{ zabbix_inventory_mode }}"
     interfaces: "{{ zabbix_agent_interfaces }}"
     visible_name: "{{ zabbix_visible_hostname|default(zabbix_agent_hostname) }}"
+    tls_psk: "{{ zabbix_agent_tlspsk_secret|default(omit) }}"
+    tls_psk_identity: "{{ zabbix_agent_tlspskidentity|default(omit) }}"
+    tls_issuer: "{{ zabbix_agent_tlsservercertissuer|default(omit) }}"
+    tls_subject: "{{ zabbix_agent_tlsservercertsubject|default(omit) }}"
+    tls_connect: "{{ zabbix_server_tlsconnect_value[zabbix_agent_tlsaccept if zabbix_agent_tlsaccept else 'unencrypted'] }}"
+    tls_accept: "{{ zabbix_server_tlsaccept_value[zabbix_agent_tlsconnect if zabbix_agent_tlsconnect else 'unencrypted'] }}"
```

Comment:
Values tls_connect and tls_accept have to be swapped. Reason: tls_accept is from the zabbix server's point of view and zabbix_agent_tlsaccept is from the agent's point of view.

Reply:
they ARE swapped ;)

Reply:
Ah yes sorry, you're right.